src/qtgui/plotter.cpp

```diff
@@ -1375,20 +1375,45 @@ void CPlotter::drawOverlay()
     {
         QList<BandInfo> bands = BandPlan::Get().getBandsInRange(m_CenterFreq + m_FftCenter - m_Span / 2,
                                                                 m_CenterFreq + m_FftCenter + m_Span / 2);
-
+        int bandLevel = 0;
+        qint64 last_max = 0;
+        qint64 last_max_0 = 0;
+        
+        std::sort(bands.begin(), bands.end(), [=]( const BandInfo& band1 , const BandInfo& band2 )->bool {
+            if (band1.minFrequency == band2.minFrequency) {
+                if (band1.maxFrequency != band2.maxFrequency) 
+                    return band1.maxFrequency > band2.maxFrequency;
+                else
+                    return band1.name < band2.name;
+            }
+            return band1.minFrequency < band2.minFrequency;
+        });
+        
         for (auto & band : bands)
         {
+            if (band.minFrequency >= last_max_0) {
+                bandLevel = 0;
+                last_max = band.maxFrequency;
+                last_max_0 = last_max;
+            } else if (band.minFrequency < last_max) {
+                --bandLevel;
+                last_max = band.maxFrequency;
+            } else if (band.minFrequency == last_max) {
+                last_max = band.maxFrequency;
+            }
```

Comment:
The `bandLevel` can reach arbitrary value, but the algorithm currently works only up to level 2 (see attached bandplan bellow to reproduce the issue). Although this will be rare case I would suggest either to limit the `bandLevel` up to some level or make this implementation generic.

There is also the edgecase where the bandplans can get out of the screen (the stack is too high). But this is really not a practical concern.

```
100000000,       102000000,       CW,  1,   #40BAFD00, Power A
100000000,       101000000,       CW,  1,   #40B00D00, Power D
101000000,       102000000,       CW,  1,   #40B00D00, Power E
100500000,       101500000,       CW,  1,   #40B00D00, Power G
```

Reply:
You are correct, it will "stair-step" for certain band plans. It still is a better implementation than no stacking at all, which simply writes one band over the top of others. It works as intended when a new band starts after the lowest level max has been exceeded.

To make it generic would require a "last_max" record for each level. Again, feasible, but is it worth the time to implement and the resources to check in order to avoid the "stair-step"?

Reply:
I see this as a great improvement. What I'm trying to achieve is to help with getting it into the codebase, in case you are willing to give it a try, I will do my best to provide support to make the implementation generic, which I believe might help.

So my opinion is it is worth the time and I'm willing to invest here.

Reply:
Cool, any hints on making it generic would be nice, but maybe the better option (at least for now) is to limit it to 2 levels? My entire bandplan (linked above) never uses more than 2 levels. What do you think is more appropriate?

Reply:
Limiting it to level 2 might also help to get into situation where the tags will go over the top of the bar (which is also unlikely to happend edgecase). Honestly both solutions (and even the staircase) are acceptable for my in the end,   so @argilo my question will be what have the better chance to get into the codebase?

The worst what can happend from my side is I will create another pull request to make the "next step ..". :)